taqueria-plugin-taquito/common.ts

```diff
@@ -0,0 +1,194 @@
+			);
+		}
+		accountKey = defaultAccount.secretKey;
+	}
+
+	const tezos = new TezosToolkit(sandbox.rpcUrl as string);
+	tezos.setProvider({ signer: new InMemorySigner(accountKey.replace(/^unencrypted:/, '')) });
+	return tezos;
+};
+
+export const configureToolKitForNetwork = async (
+	parsedArgs: UnionOpts,
+	network: NetworkConfig.t,
+	sender?: string,
+): Promise<TezosToolkit> => {
+	let account: string;
+	if (sender && sender !== TAQ_OPERATOR_ACCOUNT) {
+		const accounts = getNetworkInstantiatedAccounts(network);
+		if (accounts.hasOwnProperty(sender)) {
+			account = sender;
+		} else {
+			return sendAsyncErr(
+				`${sender} is not an account instantiated in the current environment. Check .taq/config.json`,
+			);
+		}
+	} else {
+		account = TAQ_OPERATOR_ACCOUNT;
+	}
+
+	const tezos = new TezosToolkit(network.rpcUrl as string);
+	const key = await getAccountPrivateKey(parsedArgs, network, account);
+	await importKey(tezos, key);
+	return tezos;
+};
+
+export const getDeclaredAccounts = (parsedArgs: UnionOpts): Record<string, number> =>
+	Object.entries(parsedArgs.config.accounts).reduce(
+		(acc, declaredAccount) => {
+			const alias: string = declaredAccount[0];
+			const mutez: string | number = declaredAccount[1];
+			return {
+				...acc,
+				[alias]: typeof mutez === 'string' ? parseFloat(mutez) : mutez,
+			};
+		},
+		{} as Record<string, number>,
+	);
+
+export const getSandboxInstantiatedAccounts = (sandbox: SandboxConfig.t): Record<string, SandboxAccountConfig.t> =>
+	(sandbox?.accounts)
+		? Object.entries(sandbox.accounts).reduce(
+			(acc, instantiatedAccount) => {
+				const alias: string = instantiatedAccount[0];
+				const keys = instantiatedAccount[1];
+				return alias !== 'default'
+					? {
+						...acc,
+						[alias]: keys,
+					}
+					: acc;
+			},
+			{},
+		)
+		: {};
+
+export const getNetworkInstantiatedAccounts = (network: NetworkConfig.t): Record<string, any> =>
+	network.accounts
+		? Object.entries(network.accounts).reduce(
+			(acc, instantiatedAccount) => {
+				const alias: string = instantiatedAccount[0];
+				const keys = instantiatedAccount[1];
+				return alias !== TAQ_OPERATOR_ACCOUNT
+					? {
+						...acc,
+						[alias]: keys,
+					}
+					: acc;
+			},
+			{},
+		)
+		: {};
+
+export const generateAccountKeys = async (
+	parsedArgs: UnionOpts,
+	network: NetworkConfig.t,
+	account: string,
+): Promise<void> => {
+	const tezos = new TezosToolkit(network.rpcUrl as string);
+	const key = await getAccountPrivateKey(parsedArgs, network, account);
+	await importKey(tezos, key);
+};
+
+export const handleOpsError = (err: unknown, env: string): Promise<never> => {
+	if (err instanceof Error) {
+		const msg = err.message;
+		if (/ENOTFOUND/.test(msg)) return sendAsyncErr('The RPC URL may be invalid. Check ./.taq/config.json');
```

Comment:
Just some thoughts:
Here, we swallow the error's message and return a friendly message base on matching a regex. This is all good unless there is a possible other error that matches this regex, then this behavior becomes misleading.

Maybe still returning the original error to the calling code can prevent that.

Reply:
@AlirezaHaghshenas The original error takes up the entire command screen 😞. That's why I opted for this instead. I tried to make the regexes precise enough such that it only captures the desired error. I still want to refine this part a little but I want to do that as part of the next iteration.

Reply:
Yeah, I understand that sometimes for practical reasons (like not giving the user a heart attack) we have to make a compromise.